pkg/api/api.go

```diff
@@ -12,6 +12,14 @@ type PaginationData struct {
 	Offset int `query:"offset" json:"offset"` //Offset into the total results
 }
 
+type FilterData struct {
+	Search              string `query:"search" json:"search" `                              //Search string based query to optionally filter-on
+	Arch                string `query:"arch" json:"arch" `                                  //architecture to optionally filter-on
+	Version             string `query:"version" json:"version"`                             //Version to optionally filter-on
+	AvailableForArch    string `query:"available_for_arch" json:"available_for_arch"`       //Filter by compatible arch (IE x86_64 and x86_64 or x86_64 and "Any Arch")
```

Comment:
Filter by compatible arch for client consumption (IE 'x86_64' would return Repositories with the 'x86_64' arch and Repositories where arch is not set)

Reply:
does that sound better?

Reply:
yeah I think that's more clear